qiskit/algorithms/gradients/base_qgt.py

```diff
@@ -98,8 +98,8 @@ def __init__(
         self._default_options = Options()
         if options is not None:
             self._default_options.update_options(**options)
-        self._qgt_circuit_cache = {}
-        self._gradient_circuit_cache: dict[QuantumCircuit, GradientCircuit] = {}
+        self._qgt_circuit_cache: dict[QuantumCircuit | tuple, GradientCircuit] = {}
+        self._gradient_circuit_cache: dict[QuantumCircuit | tuple, GradientCircuit] = {}
```

Comment:
shouldn't this typehint match the cache of `base_estimator_gradient.py`? they both unroll the circuit before caching.

Reply:
Correct me if I'm wrond, but here
https://github.com/Qiskit/qiskit-terra/blob/9d0ef53c99cd47d580ed8039f2469bcc3ef74cf5/qiskit/algorithms/gradients/base_qgt.py#L203-L206
tuple (return type of `_circuit_key`) is used as a key

Reply:
I see what you mean. Yeas, you're right, `QuantumCircuit` is unnecessary.